### PR TITLE
New registration page

### PR DESCRIPTION
**Pull request by Flaburgan**
Following the discussions about [the import feature](https://discourse.diasporafoundation.org/t/backup-and-restore-account-migration/544/166?u=flaburgan), here is a new registration pages, revamped, responsive, and stating to the users that they will be able to import once registered.

Capture d’écran 2021-08-29 à 23 07 30

Do you like it better with the ball?

Capture d’écran 2021-08-29 à 23 12 31

image

This PR introduces the basis of a very simple CSS framework. I feel like we don't need bootstrap anymore, it looks old and was mainly used for its grid which is obsolete now with flexboxes.

### CODE REVIEW

File: app/assets/stylesheets/framework/_framework.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+@import "fonts";
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/_framework.scss:1]: Prefer single quoted strings

File: app/assets/stylesheets/framework/_framework.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+@import "variables";
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/_framework.scss:2]: Prefer single quoted strings

File: app/assets/stylesheets/framework/_framework.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+@import "ui";
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/_framework.scss:3]: Prefer single quoted strings

File: app/assets/stylesheets/framework/_framework.scss [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+@import "layout";
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/_framework.scss:4]: Prefer single quoted strings

File: app/assets/stylesheets/framework/_framework.scss [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+@import "form";
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/_framework.scss:5]: Prefer single quoted strings

File: app/assets/stylesheets/framework/fonts.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
+@font-face {
+  font-family: 'Open Sans';
+  font-style: normal;
+  font-weight: 400;
+  src: local("Open Sans Regular"), local("OpenSans-Regular"), url(/assets/fonts/open-sans-400.woff2) format("woff2");
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/fonts.scss:5]: Prefer single quoted strings

File: app/assets/stylesheets/framework/fonts.scss [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2212, U+2215; }
+
+@font-face {
+  font-family: 'Open Sans';
+  font-style: italic;
+  font-weight: 400;
+  src: local("Open Sans Italic"), local("OpenSans-Italic"), url(/assets/fonts/open-sans-400italic.woff2) format("woff2");
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/fonts.scss:12]: Prefer single quoted strings

File: app/assets/stylesheets/framework/fonts.scss [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD; }
+
+@font-face {
+  font-family: 'Open Sans';
+  font-style: normal;
+  font-weight: 600;
+  src: local("Open Sans SemiBold"), local("OpenSans-SemiBold"), url(/assets/fonts/open-sans-600.woff2) format("woff2");
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/fonts.scss:19]: Prefer single quoted strings

File: app/assets/stylesheets/framework/form.scss [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  padding: .4rem;
+  border-radius: $radius;
+  font-size: 1rem;
+}
+
+[type=text],
+[type=email],
+[type=password] {
+  width: 320px;
+  box-shadow: inset 1px 0 1px #ddd;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/form.scss:26]: Color literals like `#ddd` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/framework/form.scss [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+@media screen and (max-width: 700px) {
+	[type=text],
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/form.scss:30]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/form.scss [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  margin-bottom: 0;
+}
+// end cleaning bootstrap
+
+button,
+input,
+select,
+textarea {
+  // Fix dirty box model rendering on inputs
+  -moz-box-sizing: border-box;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/form.scss:12]: - Properties should be ordered background-color, border, border-radius, -moz-box-sizing, -webkit-box-sizing, box-sizing, font-size, padding
- Avoid vendor prefixes.

File: app/assets/stylesheets/framework/form.scss [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+  background-color: $input-bg;
+  padding: .4rem;
+  border-radius: $radius;
+  font-size: 1rem;
+}
+
+[type=text],
+[type=email],
+[type=password] {
+  width: 320px;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/form.scss:25]: Properties should be ordered box-shadow, width

File: app/assets/stylesheets/framework/form.scss [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+}
+// end cleaning bootstrap
+
+button,
+input,
+select,
+textarea {
+  // Fix dirty box model rendering on inputs
+  -moz-box-sizing: border-box;
+  -webkit-box-sizing: border-box;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/form.scss:13]: Avoid vendor prefixes.

File: app/assets/stylesheets/framework/layout.scss [hunk 14]
```diff
@@ -0,0 +1,20 @@
+// This cleans the famous children "margin-top" problem
+.small-container::before,
+.small-container::after {
+  display: table;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/layout.scss:4]: Properties should be ordered content, display

File: app/assets/stylesheets/framework/layout.scss [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+  content: " ";
+}
+
+.small-container {
+  max-width: 800px;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/layout.scss:9]: Properties should be ordered margin, max-width

File: app/assets/stylesheets/framework/layout.scss [hunk 16]
```diff
@@ -0,0 +1,20 @@
+// This cleans the famous children "margin-top" problem
+.small-container::before,
+.small-container::after {
+  display: table;
+  content: " ";
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/layout.scss:5]: Prefer single quoted strings

File: app/assets/stylesheets/framework/ui.scss [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+	line-height: 22px;
+  padding: 8px 20px;
+	text-decoration: none;
+
+	&:disabled,
+	&:disabled:hover,
+	&.disabled,
+	&.disabled:hover {
+		background-color: #ccc;
+		color: white;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:47]: - Color `white` should be written in hexadecimal form as `#ffffff`
- Color literals like `white` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/framework/ui.scss [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+	font-weight: bold;
+	line-height: 22px;
+  padding: 8px 20px;
+	text-decoration: none;
+
+	&:disabled,
+	&:disabled:hover,
+	&.disabled,
+	&.disabled:hover {
+		background-color: #ccc;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:46]: Color literals like `#ccc` should only be used in variable declarations; they should be referred to via variable everywhere else.

File: app/assets/stylesheets/framework/ui.scss [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+input[type=submit],
+input[type=button],
+input[type=file] {
+  font-family: inherit;
+  -webkit-appearance: none;
+}
+
+.bttn {
+  border: 0;
+	border-radius: $radius;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:34]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+	cursor: pointer;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:35]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+	display: inline-block;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:36]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+	font-weight: bold;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:37]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+	line-height: 22px;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:38]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+  padding: 8px 20px;
+	text-decoration: none;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:40]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+
+	&:disabled,
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:42]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+	&:disabled:hover,
+	&.disabled,
+	&.disabled:hover {
+		background-color: #ccc;
+		color: white;
+		cursor: wait;
+	}
+
+	&:focus {
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:51]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+		outline: none;
+	}
+}
+
+.bttn-big {
+  padding: 10px 25px;
+}
+
+.bttn-primary {
+	@extend .bttn;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:61]: - Line should be indented with spaces, not tabs
- Prefer using placeholder selectors (e.g. %some-placeholder) with @extend

File: app/assets/stylesheets/framework/ui.scss [hunk 28]
```diff
@@ -0,0 +1,79 @@
...
+
+	&,
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:63]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 29]
```diff
@@ -0,0 +1,79 @@
...
+	&:visited {
+    background-color: $primary;
+		color: $primary-text;
+	}
+
+	&:hover,
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:69]: Line should be indented with spaces, not tabs

File: app/assets/stylesheets/framework/ui.scss [hunk 30]
```diff
@@ -0,0 +1,79 @@
...
+
+section {
+  margin-bottom: $double-margin;
+}
+
+button,
+input[type=submit],
+input[type=button],
+input[type=file] {
+  font-family: inherit;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:28]: Properties should be ordered -webkit-appearance, font-family

File: app/assets/stylesheets/framework/ui.scss [hunk 31]
```diff
@@ -0,0 +1,79 @@
...
+
+	&:hover,
+	&:visited:hover {
+		background: linear-gradient(-45deg, darken($primary, 4) 0%, $primary 60%);
+	}
+}
+
+.advice {
+  color: $subtle;
+  font-style: italic;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:77]: Properties should be ordered color, font-size, font-style

File: app/assets/stylesheets/framework/ui.scss [hunk 32]
```diff
@@ -0,0 +1,79 @@
...
+    font-size: 2rem;
+    margin: $default-margin 0;
+  }
+}
+
+section {
+  margin-bottom: $double-margin;
+}
+
+button,
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/ui.scss:24]: Avoid qualifying attribute selectors with an element.

File: app/assets/stylesheets/registration.scss [hunk 33]
```diff
@@ -1,54 +1,48 @@
...
-  }
+.page-registrations {
+  .fields {
+    margin: 3rem 0;
 
-  @media (max-width: $screen-xs-max) {
-    .v-center {
-      height: auto;
+    @media screen and (max-width: $breakpoint) {
+      margin: 2rem 0;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/registration.scss:8]: Properties should be ordered flex-direction, margin

File: app/assets/stylesheets/registration.scss [hunk 34]
```diff
@@ -1,54 +1,48 @@
...
 
-  .form-control.captcha-input {
-    border-bottom: 1px solid $input-border;
-    border-bottom-left-radius: 5px;
-    border-bottom-right-radius: 5px;
-    box-sizing: border-box;
-    line-height: $line-height-base;
-    padding-left: 130px;
+    .captcha-img {
+      order: 2;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/registration.scss:27]: Properties should be ordered margin-left, order

File: app/assets/stylesheets/registration.scss [hunk 35]
```diff
@@ -1,54 +1,48 @@
...
+    background-color: lighten($primary, 40); // #fff3e5;
+    border: 2px solid lighten($primary, 20); // #ffb866;
+    border-radius: $radius;
+    font-size: $smaller;
+    margin-left: $default-margin;
+    padding: $default-margin;
+    width: 240px;
+
+    @media screen and (max-width: $breakpoint) {
+      width: 100%;
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/registration.scss:43]: Properties should be ordered margin-bottom, margin-left, width

File: app/assets/stylesheets/framework/layout.scss [hunk 36]
```diff
@@ -0,0 +1,20 @@
+// This cleans the famous children "margin-top" problem
+.small-container::before,
+.small-container::after {
+	content: " ";
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/framework/layout.scss:4]: - Line should be indented with spaces, not tabs
- Prefer single quoted strings

**Comment by tclaus** [comment]
$: pronto run --unstaged -c upstream/develop

One of my favorites..

File: .scss-lint.yml [hunk 37]
```diff
@@ -188,7 +188,7 @@ linters:
 
   StringQuotes:
     enabled: true
-    style: single_quotes # or double_quotes
+    style: double_quotes
```
Comment (SuperTux88) [inline .scss-lint.yml:191]: If pronto complains a lot it isn't a solution to just change the config. If you think some of the settings don't make sense and we should change this, we can discuss this, but not as a drive-by of this PR.

Especially this one affects all existing files which previously were created with the old rule.

File: app/assets/stylesheets/framework/fonts.scss [hunk 38]
```diff
@@ -0,0 +1,20 @@
+@font-face {
```
Comment (SuperTux88) [inline app/assets/stylesheets/framework/fonts.scss:1]: I don't quite get why you need new fonts only for the registration page. I would think that the font we use on the rest of the pages also would work on a new registration page. If we want to change the font, we should change it everywhere, but again, this doesn't feel right as a drive-by with a new registration-page.

File: app/assets/stylesheets/framework/variables.scss [hunk 39]
```diff
@@ -0,0 +1,27 @@
+$breakpoint: 700px;
+
+$text: #444;
```
Comment (SuperTux88) [inline app/assets/stylesheets/framework/variables.scss:3]: This creates completely new variables which ignore all existing color and also doesn't work with the other themes.

**Review by SuperTux88 — COMMENTED** [review]
First: I like the screenshots, and I also like it with the ball (since there is space).

But I don't like the color. We don't really use this color anywhere else (except for this one place which you see once and never again), and the orange looks really strange and also looks a bit like a warning (first I thought that all fields are orange because they are empty). I would use the primary color (of the existing themes, which on the default theme is blue, on the colored themes it's the corresponding color). That way it doesn't look as warnings anymore.

The code comments aren't a full review yet, it's only what I noticed when having a quick look on it. I think if you just want to have a new registration page (which I don't have anything against it), it should work with the existing styles/themes. But the code looks more like you started to create a completely new style/theme/frontend, which I think shouldn't be done as a drive-by with a new registration page. There was a plan to create a new frontend, but I think this needs more discussion first, for example here: https://discourse.diasporafoundation.org/t/choosing-new-frontend-framework/1070 But I think that's a completely different topic/discussion.

**Comment by Flaburgan** [comment]
>But I don't like the color. We don't really use this color anywhere else (except for this one place which you see once and never again), and the orange looks really strange and also looks a bit like a warning (first I thought that all fields are orange because they are empty). I would use the primary color (of the existing themes, which on the default theme is blue, on the colored themes it's the corresponding color). That way it doesn't look as warnings anymore.

Yeah totally agree on this, I picked that color a bit randomly, we can stay with the primary blue at the moment, that is fine.

> I think if you just want to have a new registration page (which I don't have anything against it), it should work with the existing styles/themes. But the code looks more like you started to create a completely new style/theme/frontend, which I think shouldn't be done as a drive-by with a new registration page. There was a plan to create a new frontend, but I think this needs more discussion first, for example here: https://discourse.diasporafoundation.org/t/choosing-new-frontend-framework/1070 But I think that's a completely different topic/discussion.

Indeed, my idea here is to introduce a new design with an homemade minimalist CSS framework. Bootstrap is heavy, old, outdated and not needed anymore (we have flexboxes now, no need for a grid). Please note that this isn't about a new JS framework though, so IMO it is unrelated to the discourse discussion you linked.

I went with this idea of slowly introducing that new framework page per page instead of trying to rewrite everything, but maybe now (and here?) isn't the time to do that?

File: app/assets/stylesheets/framework/fonts.scss (hunk 38 above)
Comment (Flaburgan) [reply app/assets/stylesheets/framework/fonts.scss:1]: Yeah as I said I want to rework the design but I thought about going page per page, but maybe it shouldn't be part of that PR.

File: app/assets/stylesheets/framework/variables.scss (hunk 39 above)
Comment (goobertron) [reply app/assets/stylesheets/framework/variables.scss:3]: I wouldn't think the sign-up page would need to work with themes, because a not-yet-registered person won't have set a theme. But it certainly makes sense to use existing variables wherever possible, which will make changing the UI across the app easier in future.

**Comment by goobertron** [comment]
Thanks for doing this. I'm happy with the design, and it's good that it's responsive. I also question the orange, which I think is only used elsewhere for the 'getting started' banner. And if we're going to change the UI, I think it makes sense to do it in a single release for the whole UI, so I'd prefer not to introduce new typefaces just for this page. It should blend with the rest of the UI.

I only have the screen-shots to refer to, not being able to test this. Do the vertical lines to the left of the input fields disappear when valid data are entered into the fields? If not, what are they for? Their presences suggests some function, so they could be confusing if they're mere decoration.

Let me know when the text has been moved into a locale file, and I'll edit it/polish the English then.

**Comment by Flaburgan** [comment]
Okay so no more orange, and no new font: 
image

**Comment by Flaburgan** [comment]
How do you want to proceed with this idea of a custom minimalist CSS framework then? Can I start it in that PR?

**Comment by SuperTux88** [comment]
> Do the vertical lines to the left of the input fields disappear when valid data are entered into the fields? If not, what are they for? Their presences suggests some function, so they could be confusing if they're mere decoration.

I thought the same first, because the orange looked like a warning/error because it's not valid yet. But they are purely decoration, which I don't mind, as long as it's not in a warning-color (well, if a podmin creates a orange theme they might as well be orange if the whole page is orange, but not if it's the only color on the page).

> Okay so no more orange, and no new font

:+1: 

> How do you want to proceed with this idea of a custom minimalist CSS framework then? Can I start it in that PR?

We have bootstrap already anyway, so there is no harm of using it for all pages. As said, rewriting the complete frontend is a bigger task, and probably should have a bit more discussions than just as a drive-by of a new registration page.

Also as goob said "And if we're going to change the UI, I think it makes sense to do it in a single release for the whole UI, so I'd prefer not to introduce new typefaces just for this page. It should blend with the rest of the UI." it should at least be somewhat consistent, so redoing the UI is probably more something that we should start working on for 0.9. It doesn't need to be completely done for ALL the pages, and we can do it page by page on the develop branch for 0.9 once we start working on the new UI. But at least most pages should then be done before the release, so not releasing a half done new UI, and for 0.8 I think we should focus on the migration and then release and not start another big topic.

Also even if you only talk about the styles, it's probably still affected by how we are doing the new frontend, if a new frontend is done using a separate project and then only using the API, it's probably also changing how it's built. And I for example also would like to use CSS-variables instead of SCSS so color themes can be done without pre-compiling the whole styles x times.

So all that needs a bit more thinking and planning and not just start doing something as drive-by.

File: app/assets/stylesheets/framework/variables.scss (hunk 39 above)
Comment (SuperTux88) [reply app/assets/stylesheets/framework/variables.scss:3]: A podmin can set a default theme in the diaspora.toml (which I have done on my pod), and this should then still work for the registration page.

**Comment by goobertron** [comment]
> How do you want to proceed with this idea of a custom minimalist CSS framework then? 

I think a Discourse discussion would be the best place to start.

File: app/assets/stylesheets/framework/variables.scss (hunk 39 above)
Comment (goobertron) [reply app/assets/stylesheets/framework/variables.scss:3]: Ah, thanks. I didn't realise that.

**Comment by Flaburgan** [comment]
>We have bootstrap already anyway, so there is no harm of using it for all pages. As said, rewriting the complete frontend is a bigger task, and probably should have a bit more discussions than just as a drive-by of a new registration page.

Hm okay I will wait then. So I will also revert the button and the form fields still. Can I still use flexboxes instead of the bootstrap grid please? This is so much easier.

**Comment by SuperTux88** [comment]
> Can I still use flexboxes instead of the bootstrap grid please? This is so much easier.

Sure, why not :+1:

**Comment by Flaburgan** [comment]
New code using Bootstrap for fields and buttons:
image

I kept the larger font, it really was too small otherwise, and this is a complain I'm hearing a lot about diaspora* already (stuck in 13px while minimum screens are 1920px now).

**Comment by Flaburgan** [comment]
Okay, I kept the bare minimum CSS to have a design OK, new translations are added and tests are green (without having to touch them \o/). So this looks ready to review.

File: config/locales/diaspora/en.yml [hunk 40]
```diff
@@ -1091,18 +1091,23 @@ en:
 
   registrations:
     new:
+      title: "Join the community!"
```
Comment (goobertron) [inline config/locales/diaspora/en.yml:1094]: I think it's worth specifying: "Join the diaspora* community!"

File: config/locales/diaspora/en.yml [hunk 41]
```diff
@@ -1091,18 +1091,23 @@ en:
...
+      subtitle: "You're about to create an account on %{podname}. One diaspora* account is enough to reach the whole network, you don't need to create a new account here if you already have one on another pod."
```
Comment (goobertron) [inline config/locales/diaspora/en.yml:1095]: A couple of minor alterations: "You're about to create an account on %{podname}. A single account is enough to reach the whole diaspora* network. You don't need to create a new account here if you already have one on another pod."

File: config/locales/diaspora/en.yml [hunk 42]
```diff
@@ -1091,18 +1091,23 @@ en:
...
       enter_email: "Enter your email address"
-      enter_username: "Pick a username (only letters, numbers, and underscores)"
+      enter_username: "Pick a username (only letters, numbers, _ and -)"
       enter_password: "Enter a password (six character minimum)"
       enter_password_again: "Enter the same password as before"
       sign_up: "Create account"
       email: "Email"
+      email_advice: "Your e-mail will never be shared with anyone and will only be used to allow you to recover your account or send notifications."
```
Comment (goobertron) [inline config/locales/diaspora/en.yml:1102]: I suggest reordering as follows: "Your e-mail will never be shared with anyone and will only be used to send notifications and to enable you to recover your account."

File: config/locales/diaspora/en.yml [hunk 43]
```diff
@@ -1091,18 +1091,23 @@ en:
...
       username: "Username"
+      username_advice: "Your username is unique and can't be changed. It allows others to find you in the network."
```
Comment (goobertron) [inline config/locales/diaspora/en.yml:1104]: "Your username is unique and allows others to find you in the network. It can't be changed once set."

File: config/locales/diaspora/en.yml [hunk 44]
```diff
@@ -1091,18 +1091,23 @@ en:
...
       password: "Password"
       password_confirmation: "Password confirmation"
       submitting: "Submitting..."
       terms: "By creating an account you accept the %{terms_link}."
       terms_link: "terms of service"
+      import_indication: "You already have a diaspora* account and you want to migrate it on %{podname}? Please sign up first and then import your previous account from the settings."
```
Comment (goobertron) [inline config/locales/diaspora/en.yml:1110]: It's difficult to reword this and keep it simple. This is my best effort so far: "If you want to migrate  an existing diaspora* account to  %{podname}, please sign up first and then go to the user settings page to import your previous account."

**Review by goobertron — COMMENTED** [review]
With apologies for taking a long time, here are my suggestions for rewording the text.

File: config/locales/diaspora/en.yml (hunk 40 above)
Comment (Flaburgan) [reply config/locales/diaspora/en.yml:1094]: OK.

File: config/locales/diaspora/en.yml (hunk 41 above)
Comment (Flaburgan) [reply config/locales/diaspora/en.yml:1095]: Thanks, that's better.

File: config/locales/diaspora/en.yml (hunk 43 above)
Comment (Flaburgan) [reply config/locales/diaspora/en.yml:1104]: Better.

File: config/locales/diaspora/en.yml (hunk 44 above)
Comment (Flaburgan) [reply config/locales/diaspora/en.yml:1110]: Keeping the question makes it more dynamic imo. "You already have a diaspora* account and you want to migrate it on %{podname}? Please sign up first and then go to the user settings page to import your previous account."
Is that okay?

File: config/locales/diaspora/en.yml (hunk 42 above)
Comment (goobertron) [reply config/locales/diaspora/en.yml:1102]: > notifications could be seen as spam

Notifications are entirely optional, so not spam. If you are concerned about that, you could expand this to 'to send the notifications you request'. I think notifications should go first as that's the normal/most common use for the email address, whereas recovering the account is a (hopefully) rare thing. At the very least, please change 'or send notifications' at the end to 'and to send notifications'.

**Comment by Flaburgan** [comment]
Here is the last rendering with goob's changes:
image

File: config/locales/diaspora/en.yml (hunk 42 above)
Comment (Flaburgan) [reply config/locales/diaspora/en.yml]: Yeah I changed it to "Your e-mail will never be shared with anyone and will only be used to enable you to recover your account and to send notifications."

File: config/locales/diaspora/en.yml (hunk 44 above)
Comment (goobertron) [reply config/locales/diaspora/en.yml]: ""You already have a diaspora* account and you want to migrate it on %{podname}?" Just doesn't read well in English, I'm afraid. I can't think of a way of phrasing it well. Let me think again...

"Do you want to migrate an existing diaspora* account to %{podname}? If so, sign up first and then go to the user settings page to import your previous account."

That works. Does it seem dynamic enough to you?

**Comment by Flaburgan** [comment]
This is rebased now.

**Comment by Flaburgan** [comment]
I would like this to be merged because I think it really improves the registration process, explaining what is a diaspora* id etc.
If we're not sure about the migration, maybe I could strip the panel talking about it, so we can merge this MR without the migration part, and add it in a separate PR? @SuperTux88

**Comment by Flaburgan** [comment]
I removed the migration indication and rebased, this could be integrated to the next release.

**Comment by Flaburgan** [comment]
I fixed the tests. `minlength` is not fully supported for input type="password" I had to put back the `pattern` attribute as it was done before. I also added a link to the tutorials.

Here is how it looks now:

image

**Review by SuperTux88 — CHANGES_REQUESTED** [review]
The page itself looks good, but I'm still not really happy with the CSS changes.

If you want to create a new CSS framework that can also be used by other pages, then the generic variables and selectors (like `body` and `h1`) shouldn't be defined in a file called `registration.scss`.

Then the new CSS framework should be built in a way that it only affects new pages, and doesn't change font sizes and other stuff elsewhere.

And it should still work with color themes, so best use existing color variables, or if new variables are required, at least define them in a way that they can be overwritten by themes.

I'm still not really sure if all of this should be done in this PR, or be moved to a different PR. I have nothing against using flexboxes here (instead of bootstrap grid), since that's a local thing to only this page. But stuff like introducing new default styles is probably better done with a few pages directly (not necessarily all of them at once), to see what should be shared and what not. I like the idea of having a more lightweight CSS framework than using bootstrap, but I think this PR would be easier with using existing styles.

File: app/assets/stylesheets/registration.scss [hunk 45]
```diff
@@ -1,54 +1,154 @@
...
-  .ball {
-    background: image-url('branding/ball.png') no-repeat;
-    background-size: contain;
-    height: 633px;
-    max-width: 100%;
+$breakpoint: 700px;
+
+$subtle: #777;
+
+$default-size: 16px;
```
[the remark this answers is not in the document]
Comment (Flaburgan) [reply app/assets/stylesheets/registration.scss:5]: Fixed.

File: app/assets/stylesheets/registration.scss [hunk 46]
```diff
@@ -1,54 +1,154 @@
-.page-registrations {
-  .ball {
-    background: image-url('branding/ball.png') no-repeat;
-    background-size: contain;
-    height: 633px;
-    max-width: 100%;
+$breakpoint: 700px;
+
+$subtle: #777;
```
[the remark this answers is not in the document]
Comment (Flaburgan) [reply app/assets/stylesheets/registration.scss:3]: Fixed.

File: config/locales/diaspora/en.yml [hunk 47]
```diff
@@ -1091,18 +1091,20 @@ en:
...
-      enter_password: "Enter a password (six character minimum)"
-      enter_password_again: "Enter the same password as before"
+      title: "Join the diaspora* community!"
+      subtitle: "You're about to create an account on %{podname}. A single account is enough to reach the whole diaspora* network. You don't need to create a new account here if you already have one on another pod.<br />Need help? Check %{tutorials_link}."
+      tutorials_link_label: "the tutorials"
       sign_up: "Create account"
       email: "Email"
+      email_advice: "Your e-mail will never be shared with anyone and will only be used to enable you to recover your account and to send notifications."
       username: "Username"
+      username_advice: "Your username is unique and allows others to find you in the network. It can't be changed once set. Only letters, numbers, _ and -."
```
[the remark this answers is not in the document]
Comment (Flaburgan) [reply config/locales/diaspora/en.yml:1101]: Fixed.

File: app/assets/stylesheets/registration.scss [hunk 48]
```diff
@@ -1,54 +1,154 @@
...
+$double-margin: $default-margin * 2;
+
+$radius: 4px;
+
+
+// UI, maybe extracted to be shared with other pages later
+// New default fontsize the page
+html,
+body {
+  font-size: 16px;
```
[the remark this answers is not in the document]
Comment (Flaburgan) [reply app/assets/stylesheets/registration.scss]: Fixed.

**Comment by Flaburgan** [comment]
I rework my changes, is it better like that?

**Comment by Flaburgan** [comment]
@SuperTux88 I rework my changes, is it better like that?

File: app/views/registrations/_new.erb [hunk 49]
```diff
@@ -0,0 +1,77 @@
...
+    </header>
+    <%= form_for(resource, url: registration_path(resource_name)) do |f| %>
+      <div class="flex-container fields">
+        <div>
+          <section>
+            <p>
+              <label for="user_username"><%= t("registrations.new.username") %></label>
+            </p>
+            <p>
+              <input type="text" id="user_username" name="user[username]" value="<%= @user.username %>" class="form-control" pattern="[A-Za-z0-9_.\-]+" required />@<%= AppConfig.pod_uri.host %>
```
Comment (SuperTux88) [inline app/views/registrations/_new.erb:15]: Not a problem on most production pods, but I just noticed that on my local dev-system, this doesn't include the port, which would also be part of the diaspora-ID. What you want to use instead is `AppConfig.bare_pod_uri`, or maybe even `User.diaspora_id_host` (which also includes the `@`).

File: app/assets/stylesheets/registration.scss [hunk 50]
```diff
@@ -1,54 +1,79 @@
...
-  .ball {
-    background: image-url('branding/ball.png') no-repeat;
-    background-size: contain;
-    height: 633px;
-    max-width: 100%;
+  // For the mobile version which doesn't have the same global CSS
+  background: $body-bg;
+
+  #app {
+    margin: -10px;
```
Comment (SuperTux88) [inline app/assets/stylesheets/registration.scss:6]: On mobile, there is a very small horizontal scrollbar, and I think it has something to do with this, since this makes the page actually 10px bigger (since it increases the size of `#app`), so what you want to do instead is probably reduce the `padding` of `#main` and leave `#app` as it is.

File: app/assets/stylesheets/registration.scss [hunk 51]
```diff
@@ -1,54 +1,79 @@
...
+  .captcha {
+    align-items: center;
+    display: flex;
 
-    h1 {
-      font-size: 35px;
-      margin: 12px 0;
+    .captcha-img {
+      margin-left: 5px;
+      order: 2;
```
Comment (SuperTux88) [inline app/assets/stylesheets/registration.scss:42]: I don't know if this is intended, but this only works in the "desktop" (default) view, but not in "mobile" view. Since both should use the same responsive design, I assume this should look always the same on small screens, independent of in which mode you actually are? So you probably just want to delete `app/views/simple_captcha/_simple_captcha.mobile.haml` so it uses the same template as in "desktop" view?

**Review by SuperTux88 — CHANGES_REQUESTED** [review]
Thanks, yes this looks already a lot better. :+1:  But I still found a few small things.

By the way, you don't need to respond to each of my comments, it's enough if you write a comment if you are done with everything and I also receive emails when you push new stuff. If you really want to react to the specific comments to mark that you have done them, you can maybe just react with a :+1: to the comment, this doesn't send a separate email for each comment to all the watchers of the repo.

File: app/assets/stylesheets/registration.scss (hunk 50 above)
Comment (Flaburgan) [reply app/assets/stylesheets/registration.scss:6]: Ah nice catch, this is a regression on the last change I made. Fixed, thanks for noticing it.

**Review by SuperTux88 — APPROVED** [review]
